File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> ingredient) {
    int answer = 0;
    
    vector<int> v;
    for (auto num: ingredient) {
        v.push_back(num);
        
        vector<int> sub(v.end() - 4, v.end());
        if (sub == vector<int>{1, 2, 3,1 }){
            answer += 1;
            v.erase(v.end() - 4, v.end());
        }
    }
    
    return answer;
}

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.py
def solution(ingredient):
    answer = 0
    
    li = []
    
    for num in ingredient:
        li.append(num)
        if (li[-4:] == [1, 2, 3, 1]):
            answer += 1
            for i in range(4):
                li.pop()
    
    return answer